From 2a181956f380bea352b79c15a0b83d28e8eaca26 Mon Sep 17 00:00:00 2001
From: CHUNGPT17 <chung24091990@gmail.com>
Date: Sun, 26 Jan 2025 05:36:58 +0000
Subject: [PATCH] 900-arm64-boot-add-dts-files

---
 arch/arm64/boot/dts/rockchip/Makefile | 1 ++++
 1 file changed, 1 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/Makefile b/arch/arm64/boot/dts/rockchip/Makefile
index d5aefff..31977f7 100644
--- a/arch/arm64/boot/dts/rockchip/Makefile
+++ b/arch/arm64/boot/dts/rockchip/Makefile
@@ -55,3 +55,4 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-sapphire.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-sapphire-excavator.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399pro-rock-pi-n10.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-evb1-v10.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3568-nanopi-r5s.dtb
-- 
2.45.2

